table_engine/src/partition/mod.rs

```diff
@@ -101,42 +125,106 @@ impl From<meta_pb::Definition> for Definition {
     }
 }
 
-impl From<PartitionInfo> for meta_pb::add_table_meta::PartitionInfo {
+impl From<PartitionInfo> for meta_pb::PartitionInfo {
     fn from(partition_info: PartitionInfo) -> Self {
         match partition_info {
-            PartitionInfo::Hash(v) => Self::Hash(meta_pb::HashPartitionInfo {
-                definitions: v.definitions.into_iter().map(|v| v.into()).collect(),
-                expr: v.expr.to_vec(),
-                linear: v.linear,
-            }),
-            PartitionInfo::Key(v) => Self::KeyPartition(meta_pb::KeyPartitionInfo {
-                definitions: v.definitions.into_iter().map(|v| v.into()).collect(),
-                partition_key: v.partition_key,
-                linear: v.linear,
-            }),
+            PartitionInfo::Hash(v) => meta_pb::PartitionInfo {
+                partition_info_enum: Some(meta_pb::partition_info::PartitionInfoEnum::Hash(
+                    meta_pb::HashPartitionInfo {
+                        definitions: v.definitions.into_iter().map(|v| v.into()).collect(),
+                        expr: Bytes::into(v.expr),
+                        linear: v.linear,
+                    },
```

Comment:
I think we should build the `HashPartitionInfo` first, and then pass here.
Usually, it's not suggested to place too many code in the initialization.